### PR TITLE
Filter hidden files

### PR DESCRIPTION
**Pull request by aduffeck**
With this PR hidden files (dotfiles) will no longer be part of the search results.

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/16746/42/1) failed. Further test are cancelled...

**Comment by mmattel** [comment]
Just a question, should this be set by default or maybe defined via a env var? Because you do not restrict searching but filtering the results?

**Comment by aduffeck** [comment]
@mmattel the e2e test suite also wasn't happy about filtering hidden files by default. We'll now index the hidden state so it's available for future changes but we don't filter by default. So the search behaves exactly the same as before.

**Review by butonic — APPROVED** [review]

File: services/search/pkg/search/index/index_test.go [hunk 1]
```diff
@@ -176,6 +176,15 @@ var _ = Describe("Index", func() {
 				assertDocCount(ref.ResourceId, `Name:1234*`, 1)
 			})
 
+			It("filters hidden files", func() {
+				ri.Path = ".hidden.pdf"
+				ref.Path = "./" + ri.Path
+				err := i.Add(ref, ri)
+				Expect(err).ToNot(HaveOccurred())
+
+				assertDocCount(ref.ResourceId, `Name:*hidden* +Hidden:true`, 0)
```
Comment (butonic) [inline services/search/pkg/search/index/index_test.go:185]: shouldnt this be 1?

**Review by butonic — CHANGES_REQUESTED** [review]

**Review by butonic — APPROVED** [review]

**Pull request merged**